### PR TITLE
Bump up Expo SDK to 50.0.3

### PR DESCRIPTION
**Pull request by kowczarz**
### Details

We had to bump up Expo SDK, because in latest version it fixes autolinking bug https://github.com/expo/expo/pull/26497 which is a blocker of https://github.com/Expensify/App/pull/30829. The reason I'm updating it in a separate PR is that it will be easier to cherrypick the change to Fabric Migration PR https://github.com/Expensify/App/pull/13767.

### Fixed Issues

Needed for https://github.com/Expensify/App/issues/20471

### Tests

- [x] App starts up correctly on every supported platform
- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
        - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Screenshot 2024-01-26 at 16 08 51

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Screenshot 2024-01-26 at 15 43 29

</details>

<details>
<summary>iOS: Native</summary>

Simulator Screenshot - iPhone 15 Pro - 2024-01-26 at 15 42 31

</details>

<details>
<summary>iOS: mWeb Safari</summary>

Simulator Screenshot - iPhone 15 Pro - 2024-01-26 at 15 41 30

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-01-26 at 15 37 58

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-01-26 at 15 45 17

</details>

### CODE REVIEW

**Comment by kowczarz** [comment]
@francoisl fair point I forget to commit `Podfile.lock` changes, but I'm not sure if we want to commit your changes of `Podfile`, because it looks like we already have that lines in our code https://github.com/Expensify/App/blob/main/ios/Podfile#L69-L73

**Comment by francoisl** [comment]
I can't build Android locally (though I think my whole dev environment is kind of broken) so I'm trying to use an ad-hoc build, but it's failing with:

```
CommandError: It looks like you're trying to use web support but don't have the required
dependencies installed.

Please install @expo/metro-runtime@~3.1.1 by running:

npx expo install @expo/metro-runtime
```

iOS is also failing, logs can be found here: https://github.com/Expensify/App/actions/runs/7701854602

@c3024 are you able to build on Android and iOS locally?

**Comment by kowczarz** [comment]
Ok, I will check the problem and I will probably need to update the CI workflow. I'll reach @mountiny  to run the build to test if my fixes work, because he's in same timezone and we won't waste time waiting to sync.

**Comment by francoisl** [comment]
Sounds good, thanks. I'll see if I can find anything to fix it today as well.

Also, feel free to ask any engineer to start an ad-hoc build for you in Slack if Vit isn't available (although, he seems to be online almost 24 hours a day 😄)

**Comment by mountiny** [comment]
> although, he seems to be online almost 24 hours a day 😄

😶‍🌫️😶‍🌫️😶‍🌫️

**Comment by c3024** [comment]
Android build worked for me


https://github.com/Expensify/App/assets/102477862/4c758634-7a00-4ec0-ad46-7b0c0b702bd2

**Comment by mountiny** [comment]
https://github.com/Expensify/App/actions/runs/7710101028

**Comment by c3024** [comment]
iOS build failed

Screenshot 2024-01-30 at 4 50 34 PM

**Comment by c3024** [comment]
iOS - Installed node_modules afresh and ran pod install and built the app. Failed again.

Screenshot 2024-01-30 at 4 59 57 PM

**Comment by kowczarz** [comment]
@c3024 I've merged latest `main` please check now if you can run the iOS.

**Comment by c3024** [comment]
Builds well on iOS 👍 

https://github.com/Expensify/App/assets/102477862/91d26c97-6e4f-4a48-93fc-9cf2b407747e

**Comment by c3024** [comment]
Desktop failing 😞 

Screenshot 2024-01-30 at 7 37 25 PM

**Comment by kowczarz** [comment]
It seems there was a problem with GitHub actions runner, because the iOS build just timed out without any error.

**Review by c3024 — APPROVED** [review]
Working well on all platforms. 🚀

**Review by francoisl — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -99,7 +100,7 @@
     "dom-serializer": "^0.2.2",
     "domhandler": "^4.3.0",
     "expensify-common": "git+ssh://git@github.com/Expensify/expensify-common.git#4a61536649cbfe49236a35bc7542b5dfd0767e4a",
-    "expo": "^50.0.0-preview.7",
+    "expo": "^50.0.3",
     "expo-image": "1.10.1",
```
Comment (shubham1206agra) [inline package.json:104]: Why this was not bumped in this PR?

**Pull request merged**